DllImportGenerator/DllImportGenerator/Marshalling/ICustomNativeTypeMarshallingStrategy.cs

```diff
@@ -874,22 +928,27 @@ private StatementSyntax GenerateContentsMarshallingStatement(TypePositionInfo in
                 NativeIndex = info.NativeIndex
             };
 
-            StatementSyntax marshallingStatement = Block(
-                List(elementMarshaller.Generate(
-                    localElementInfo,
-                    elementSubContext)));
+            List<StatementSyntax> elementStatements = elementMarshaller.Generate(localElementInfo, elementSubContext).ToList();
 
-            if (elementMarshaller.AsNativeType(elementInfo) is PointerTypeSyntax)
+            if (elementStatements.Count != 0)
```

Comment:
One of the few uses of Linq I like is the `elementStatements.Any()` &ndash; up to you.